Remove VersionAbstractionFedoraObjectHandler and Fedora2-specific code

bcail opened the pull request:
Remove VersionAbstractionFedoraObjectHandler and Fedora2-specific code
* * *

**JIRA Ticket**: (None)

# What does this Pull Request do?
Removes Fedora2-specific content, and collapses 3 object handler interface/classes down to 2.

# How should this be tested?
Automated tests should pass. Any migrations you run (except from a Fedora2 repo) should run the same as previously.

# Interested parties
@fcrepo/committers

dbernstein commented:
@bcail :  So if I'm grokking your changes what you've done is move version handling into from VersionAbstractionFedoraObjectHandler into ObjectAbstractionStreamingFedoraObjectHandler so that you can now get both object and version information in one pass.  Is that right?   Just want to make sure I understand the motivation for that particular refactoring.

dbernstein reviewed (commented):
Looks good over all.  See my other comment.

bcail commented:
@dbernstein thanks for looking at this. Basically, I didn't see the need for 3 separate handlers, so I removed one. I think it would work to remove another handler, too, but that might be more work. There's also code/config related to fedora-2 and fedora-2-style disseminators I removed.
